src/runtime/pickle_runner.ts

```diff
@@ -375,9 +411,36 @@ export default class PickleRunner {
         },
       })
     }
-    return await this.invokeStep(
-      testStep.pickleStep,
-      testStep.stepDefinitions[0]
+
+    const beforeStepHooksResult = await this.runStepHooks(
+      this.getBeforeStepHookDefinitions(),
+      null
     )
+    let finalStepResult = new Query().getWorstTestStepResult(
+      beforeStepHooksResult
+    )
+    if (finalStepResult.status !== Status.FAILED) {
+      const stepResult = await this.invokeStep(
+        testStep.pickleStep,
+        testStep.stepDefinitions[0]
+      )
+      const afterStepHooksResult = await this.runStepHooks(
+        this.getAfterStepHookDefinitions(),
+        stepResult
+      )
+
+      const stepHookResults = beforeStepHooksResult.concat(afterStepHooksResult)
+      finalStepResult = stepResult
+      finalStepResult.status = new Query().getWorstTestStepResult(
+        stepHookResults.concat(stepResult)
+      ).status
+      for (const stepHookResult of stepHookResults) {
+        finalStepResult.duration = addDurations(
+          finalStepResult.duration,
+          stepHookResult.duration
+        )
+      }
```

Comment:
This looks like we only sum the durations if all beforeHooks passed and we should sum it no matter what.